pkg/util/cluster_files/cluster_files.go

```diff
@@ -0,0 +1,100 @@
+package cluster_files
+
+import (
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/admiral/pkg/log"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+)
+
+var (
+	UnsupportedSchema   error = errors.New("The schema is not supported by cluster_files module")
+	MalformedURL        error = errors.New("The url is not well formed for cluster_files module")
+	FileContentNotFound error = errors.New("File data not found in object")
+)
+
+// Get retrieves a config from a secret, configmap or file within the k8s cluster
+// using an url schema that supports configmap://<namespace>/<configmap-name>/<data-file>
+// secret://<namespace>/<secret-name>/<data-file> and file:///<path> returning
+// a local path to the file
+func Get(k8sClient kubernetes.Interface, urlAddress string) (pathStr string, err error) {
+	klog.V(log.DEBUG).Infof("reading cluster_file: %s", urlAddress)
+	parsedUrl, err := url.Parse(urlAddress)
+	if err != nil {
+		klog.Error(err)
```

Comment:
We're returning the error so no need to log.

Reply:
oops, an artifact from when I was debugging.

Reply:
I assume you intended to remove it 😄